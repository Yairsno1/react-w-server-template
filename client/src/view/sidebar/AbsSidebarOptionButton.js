import React, {Component} from 'react';

class AbsSidebarOptionButton extends Component {
  render() {
    const txtColor = this.props.textColor;
    const classItems = [
      'w3-button',
      'hp-btn',
      'w3-block',
      'w3-padding-small',
      'w3-text-' + txtColor,
      'w3-hover-' + txtColor,
      'w3-hover-text-black'];

    return (
      <button className={classItems.join(' ')} onClick={this.props.onSelected}>
        {this.props.children}
      </button>
    );
  }
}

export default AbsSidebarOptionButton;
