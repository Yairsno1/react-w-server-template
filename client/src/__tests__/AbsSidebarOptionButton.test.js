//Testing React component that compose children props
//Jest-Enzyme matchers
//https://github.com/FormidableLabs/enzyme-matchers/blob/master/README.md

import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import AbsSidebarOptionButton from '../view/sidebar/AbsSidebarOptionButton';

let m_color;
let m_pText;
let m_children;

beforeAll(() => {
  m_color = 'red';
  m_pText = 'Woodoo child';
  m_children = <p>{m_pText}</p>;
});

it('AbsSidebarOptionButton_rendersWithoutCrashing', () => {
  shallow(
    <AbsSidebarOptionButton textColor={m_color}>
      {m_children}
    </AbsSidebarOptionButton>
  );
});

it('AbsSidebarOptionButton_snapshot', () => {
  const btn = renderer.create(
    <AbsSidebarOptionButton textColor={m_color}>
      {m_children}
    </AbsSidebarOptionButton>
  );
  let actualBtn = btn.toJSON();

  expect(actualBtn).toMatchSnapshot();
});

it('AbsSidebarOptionButton_textColor', () => {
  const expectedTextColor = 'w3-text-' + m_color;

  const optionBtn = shallow(
    <AbsSidebarOptionButton textColor={m_color}>
      {m_children}
    </AbsSidebarOptionButton>
  );

  const actualDOMBtn = optionBtn.find('button');

  expect(actualDOMBtn).toHaveClassName(expectedTextColor);
});

it('AbsSidebarOptionButton_hoverColor', () => {
  const expectedHoverColor = 'w3-hover-' + m_color;

  const optionBtn = shallow(
    <AbsSidebarOptionButton textColor={m_color}>
      {m_children}
    </AbsSidebarOptionButton>
  );

  const actualDOMBtn = optionBtn.find('button');

  expect(actualDOMBtn).toHaveClassName(expectedHoverColor);
});

it('AbsSidebarOptionButton_children', () => {
  const expectedChildren = m_children;

  const optionBtn = mount(
    <AbsSidebarOptionButton textColor={m_color}>
      {m_children}
    </AbsSidebarOptionButton>
  );

  expect(optionBtn).toContainReact(expectedChildren);
});
