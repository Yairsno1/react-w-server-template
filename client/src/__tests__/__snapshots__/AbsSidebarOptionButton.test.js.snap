// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AbsSidebarOptionButton_snapshot 1`] = `
<button
  className="w3-button hp-btn w3-block w3-padding-small w3-text-red w3-hover-red w3-hover-text-black"
>
  <p>
    Woodoo child
  </p>
</button>
`;
